framework/decode/dx12_dump_resources.cpp

```diff
@@ -60,31 +68,33 @@ void Dx12DumpResources::StartDump(ID3D12Device* device, const std::string& filen
     // prepare for output data
     json_options_.format = kDefaultDumpResourcesFileFormat;
 
-    json_filename_    = filename;
+    json_filename_    = capture_file_name;
     auto ext_pos      = json_filename_.find_last_of(".");
     auto path_sep_pos = json_filename_.find_last_of(util::filepath::kPathSepStr);
     if (ext_pos != std::string::npos && (path_sep_pos == std::string::npos || ext_pos > path_sep_pos))
     {
         json_filename_ = json_filename_.substr(0, ext_pos);
     }
-    json_filename_ += "_resources_submit_" +
-                      std::to_string(track_dump_resources_.target.dump_resources_target.submit_index) + "_command_" +
-                      std::to_string(track_dump_resources_.target.dump_resources_target.command_index) + "_drawcall_" +
-                      std::to_string(track_dump_resources_.target.dump_resources_target.drawcall_index) + "." +
-                      util::get_json_format(json_options_.format);
-
-    json_options_.data_sub_dir = util::filepath::GetFilenameStem(json_filename_);
+    json_options_.data_sub_dir = json_filename_;
     json_options_.root_dir     = util::filepath::GetBasedir(json_filename_);
+    json_filename_ += "_dr." + util::get_json_format(json_options_.format);
```

Comment:
The line
`json_filename_ += "_dr." + util::get_json_format(json_options_.format);`
Should go above the line
`json_options_.data_sub_dir = util::filepath::GetFilenameStem(json_filename_);`

Otherwise capture filename parts may get cut off. For example:
`CapturedApp.exe_20240625T123147_dr_shader_resourceres_id_28_sub_0_before.bin`
becomes
`CapturedApp_shader_resourceres_id_28_sub_0_before.bin`
when the capture file is named `CapturedApp.exe_20240625T123147.gfxr`